assets/js/item_details.js

```diff
@@ -149,6 +177,110 @@ $("#add-image-input").change(function (e) {
 });
 
 
+$(".deleteRow").click((e) => {
+    if ($(e.target).siblings(".item_quantity").val() > 0) {
+        return
+    }
+    let deletedLocation = $(e.target).parent().parent().parent()
+    let deletedId = deletedLocation.attr("id")
+    let deletedName = $(deletedLocation).children("th").text()
+    deletedLocation.replaceWith("<input hidden name=\"deletedRows\" value=\""+ deletedId +"\">\n")
+
+    // Add back location to selector box after removal
+    let locationSelector = $("#locationSelect")
+    let locationSelectorAdd = "<option data-id=\""+deletedId+"\">"+ deletedName +"</option>"
+    locationSelector.append(locationSelectorAdd)
+
+    // show the location selector if there are locations to be shown
+    if (locationSelector.length > 0) {
+        $(".locationAdder").show()
+    }
+
+
+})
+
+
+$(".addRow").click(function () {
+    let locationSelector = $("#locationSelect option:selected");
+
+    let locationId = locationSelector.attr('data-id');
+    let locationName = locationSelector.val()
+    // This value is 
+    let itemStorageTag = "newItemStorage"
+
+    let newRow = $($.parseHTML(
```

Comment:
I'm not going to make you change this, but next time, if you wanted, you could make a hidden element  in the html and just clone it every time you want a new one with $(object).clone()